app/mockServices/mockProjects.service.ts

```diff
@@ -0,0 +1,52 @@
+import * as angular from 'angular';
+
+interface IProjectsService {
+  canCreate(): angular.IPromise < any >;
+  get(projectName: string): angular.IPromise < any >;
+  update(projectName: string, data: any): angular.IPromise < any >;
+}
+
+export class ProjectsService implements IProjectsService {
+  public static $inject = ['$q'];
+
+  private $q : any;
+
+  constructor ($q: angular.IQService) {
+    this.$q = $q;
+  }
+
+  public canCreate() {
+    let deferred = this.$q.defer();
+
+    setTimeout(() => {
```

Comment:
I've seen usage of $timeout and setTimeout() in origin-web-catalog, do we have a preference/best practice?